main.go

```diff
@@ -193,6 +200,33 @@ func main() {
 			setupLog.Error(err, "unable to create controller", "controller", "AzureCluster")
 			os.Exit(1)
 		}
+		// TODO(ace): uncomment this when CAPI upstream supports AKS feature flag, or we can have our own
+		// if feature.Gates.Enabled(feature.AKS) {
```

Comment:
Probably want MachinePool flag AND AKS feature flag.

Reply:
I also need to define the feature flags in CAPZ rather than import from CAPI if we want to keep this CAPZ specific. I clarified this with @vincepri in slack